### PR TITLE
Update go to version 1.20

### PR DESCRIPTION
**Pull request by janboll**

### CODE REVIEW

File: pkg/pgp/pgp_test.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func TestTestEncryptNoEncryptionKey(t *testing.T) {
 	assert.Nil(t, err)
 	err = TestEncrypt(entity)
 	assert.NotNil(t, err)
-	assert.Regexp(t, `cannot encrypt a message .+ no encryption keys`, err)
+	assert.Regexp(t, `error setting up encryption for PGP message: openpgp: invalid argument: cannot encrypt a message to key id .+ because it has no valid encryption keys`, err)
```
Comment (kwilczynski) [inline pkg/pgp/pgp_test.go:99]: Nice update! From (version) 2021 to 2023!

Comment (janboll) [reply pkg/pgp/pgp_test.go:99]: Should be fine, since we don't rely on terraform encryption, right?

Comment (kwilczynski) [reply pkg/pgp/pgp_test.go:99]: Yes, it should be no issue.

**Review by patjlm — APPROVED** [review]

**Pull request merged**